### PR TITLE
feat: new pkgdb realise subcommand

### PR DESCRIPTION
**Pull request by limeytexan**
## Proposed Changes

This commit adds a new `pkgdb realise` subcommand for realising pkgs referenced from within a manifest lockfile. It will be used in conjunction with the nix-based `buildenv` as detailed in a following PR.

The tests added with this commit are copied largely from the corresponding tests for `pkgdb buildenv`, with the expectation that we will be removing all code and tests associated with this pkgdb subcommand in future.

## Release Notes

N/A

### CODE REVIEW

**Review by mkenigs — COMMENTED** [review]
nit: can we rename it to `pkgdb realise`?

**Comment by limeytexan** [comment]
> nit: can we rename it to `pkgdb realise`?

The code is patterned after `buildenv`, which is a concatenation of `<verb><noun>`, and this naming found its way into the code with varying capitalisation of `{verbnoun,Verbnoun,verbNoun,VerbNoun}`. These classes are what drives the arg parsing and subcommand structure, so the fastest and easiest way to write the code was to come up with a similar concatenation, so that's why I chose `realisepkgs`.

I literally don't care what we call it, but I do care to avoid unnecessary work, particularly as this code isn't at all user facing and isn't expected to live long. I'd say if you can change it in 10 minutes then do that, but otherwise I really don't think it's worth it.

**Comment by limeytexan** [comment]
I attempted to rename the `realisepkgs` class to `realise` and rename all the associated files, but the latter clashed with the existing `realise.{cc,hh}` files. So I looked for the minimal change to trick the parser and this 2-line fix seems to do the trick. So with this the classes still bear the name(s) "[rR]ealise[pP]kgs" but the subcommand is now the [mismatched] "realise".

File: pkgdb/src/main.cc
Comment (ysndr) [inline pkgdb/src/main.cc:133]: suggestion(nb):

```suggestion
  if ( prog.is_subcommand_used( cmdRealisePkgs.getParser() ) )
  { 
    return cmdRealisePkgs.run();
  }
```

File: pkgdb/tests/data/realise/lockfiles/single-package/manifest.lock
Comment (ysndr) [inline pkgdb/tests/data/realise/lockfiles/single-package/manifest.lock:1]: I think we are actively removing support for v0 manifests and lockfiles so its a bit surprising to see `"lockfile-version": 0` being used here.

(non-blocking)
My suggestion would be to upgrade this to v1, now or in a following iteration.

File: pkgdb/src/realisepkgs/command.cc
Comment (ysndr) [inline pkgdb/src/realisepkgs/command.cc:61]: Note: the current buildenv command carries information about the original package id and priorities much further.
IMO dropping this would be a functional regression, if only for the weaker error messaging and lost support for priorities.

**Comment by limeytexan** [comment]
I did some further renaming of `realisepkgs` -> `realise` in the pkgdb tests directory, both for the command invocations and also the associated test data. With that this PR now passes tests for the `pkgdb realise` subcommand. With an approval we can get that squashed into a single commit and added to the merge queue.

**Review by ysndr — APPROVED** [review]
comment:
This PR looks like it could have been a refactoring of `flox::buildenv::createFloxEnv` specifically the package loop here: https://github.com/flox/flox/blob/6cbfd8d22243d82849790f2ba2ca8b6e852b632b/pkgdb/src/buildenv/realise.cc#L903-L919
plus a subcommand to call this.

Instead all of build env was basically duplicated.
I see the appeal to be able to eventually just delete the buildenv packages as a whole, but its also harder to see the changes made against int in this format.

File: pkgdb/src/realisepkgs/command.cc
Comment (limeytexan) [reply pkgdb/src/realisepkgs/command.cc:61]: That support will be provided bu the derivation-based buildenv.

File: pkgdb/tests/data/realise/lockfiles/single-package/manifest.lock
Comment (limeytexan) [reply pkgdb/tests/data/realise/lockfiles/single-package/manifest.lock:1]: Agreed; I literally copied all test data from the buildenv files and modified as required, and this overlapped Zach's efforts to retire all v0 test data. We will need to similarly re-sweep that test data once this gets merged.

**Comment by limeytexan** [comment]
@ysndr I applied your suggestion, squashed all commands and now re-running tests. When that finishes I'm hoping we'll be good to merge - let me know if you see any other blockers.

File: pkgdb/src/realisepkgs/command.cc
Comment (ysndr) [reply pkgdb/src/realisepkgs/command.cc:61]: derivation or store-path based?
Also won't that still need to know the priority of the store paths and which package (id) the path is associated with?

**Pull request merged**
